doc: improve ninja build for --built-in-modules-path (#53007)
nodejs/node: doc/contributing/building-node-with-ninja.md

[Ninja][] is supported in the Makefile. Run `./configure --ninja` to configure
the project to run the regular `make` commands with Ninja.

> If you wish to only modify JS layer in `lib`, you can use:

Reviewer: ```suggestion
When modifying only the JS layer in `lib`, you can use:
```

         # regardless of the number of cores on the current machine.
```

If you wish to only modify JS layer in `lib`, you can use:

```bash
./configure --ninja --node-builtin-modules-path "$(pwd)"
```

> 

Reviewer: I would move this so it appears before the `make -j4` command – ideally just after the paragraph that mention `./configure`

Author: thanks for the suggestion, I have moved this to the `./configure` section.
